st2actions/st2actions/runners/actionchainrunner.py

```diff
@@ -63,6 +67,35 @@ def init_vars(self, action_parameters):
             self.vars = self._get_rendered_vars(self.actionchain.vars,
                                                 action_parameters=action_parameters)
 
+    def validate(self):
```

Comment:
To add to this - that's something we should agree on and consistently follow for all the resources and use cases.

Imo, ideally we would always try to perform as much validation as possible as early as possible (this means during create / load time). This provides a better user experience since user doesn't need to wait for the run time to get back feedback.